controllers/vspherecluster_controller.go

```diff
@@ -141,25 +149,13 @@ func (r *VSphereClusterReconciler) reconcileNormal(ctx *context.ClusterContext)
 			ctx.VSphereCluster.Namespace, ctx.VSphereCluster.Name)
 	}
 
-	// If the VSphereCluster resource has no API endpoints set then requeue
-	// until an API endpoint for the cluster resource can be found.
-	if len(ctx.VSphereCluster.Status.APIEndpoints) == 0 {
```

Comment:
Hi @yastij,

Because it matters absolutely nothing at all if the API endpoints are set if the next call can't get a connection to the target cluster. Thus a failure is fine, because nothing proceeds until the cloud provider secret can be created.

Reply:
ack